target/bin/addmailuser

```diff
@@ -44,7 +44,11 @@ PASSWD="${*}"
 
 # Protect config file with lock to avoid race conditions
 touch "${DATABASE}"
-create_lock "$(basename "$0")"
+SCRIPT_NAME="$(basename "$0")"
+LOCK_ID="${RANDOM}"
```

Comment:
Why not move this two lines into the create_lock function?

Reply:
Both are unique to the script executing the create and remove_lock functions. Maybe I misunderstand but I think if I move them into the function SCRIPT_NAME would be `helper-functions.sh` and the LOCK_ID wouldn't be able to be set with the ID for the specific instance of the script.

Reply:
Your statement would be true, if `helper-functions.sh` would be called directly.
But in this case, `helper-functions.sh` is sourced. You can consider its content as part of the current file, therefore no problems with `$0` etc.

Reply:
> Both are unique to the script executing the create and remove_lock functions.

I believe the question was due to these vars not being used anywhere else in the script, they're serving this purpose for locking only and the repetition in other files establishes a pattern that could be DRY instead.

> SCRIPT_NAME would be `helper-functions.sh`

Not if you pass `"$0"` as the arg, or `"$(basename "$0")"` if that's not the case?

> LOCK_ID wouldn't be able to be set with the ID for the specific instance of the script.

It's a random value that is used on both `create_lock` and the trap for `remove_lock`, if those are both wrapped into a helper function as well, that should be a non-issue?

There is one other usage so far where you avoid `create_lock` early and use the values later that would not utilize this helper/wrapper method I guess.

Reply:
> Your statement would be true, if `helper-functions.sh` would be called directly.
> But in this case, `helper-functions.sh` is sourced. You can consider its content as part of the current file, therefore no problems with `$0` etc.

Oh DERP! You're right. Silly mistake. Thanks for catching that

Reply:
> > Both are unique to the script executing the create and remove_lock functions.
> 
> I believe the question was due to these vars not being used anywhere else in the script, they're serving this purpose for locking only and the repetition in other files establishes a pattern that could be DRY instead.
> 
> > SCRIPT_NAME would be `helper-functions.sh`
> 
> Not if you pass `"$0"` as the arg, or `"$(basename "$0")"` if that's not the case?
> 
> > LOCK_ID wouldn't be able to be set with the ID for the specific instance of the script.
> 
> It's a random value that is used on both `create_lock` and the trap for `remove_lock`, if those are both wrapped into a helper function as well, that should be a non-issue?
> 
> There is one other usage so far where you avoid `create_lock` early and use the values later that would not utilize this helper/wrapper method I guess.

I don't remember why, but I thought I removed the trap from create_lock because it was breaking something. We should set it in there and not have to call it from the various scripts. I'll add it back and test everything.